internal/agent/agent.go

```diff
@@ -169,14 +197,65 @@ func (a *Agent) RunPoll(ctx context.Context, wg *sync.WaitGroup) {
 			logger.Log.Debug("worker is shutdown", zap.String("name", "poll"))
 			return
 		default:
-			logger.Log.Debug("message from worker", zap.String("name", "poll"))
 			a.loadMetrics()
+			logger.Log.Debug("message from worker", zap.String("name", "poll"), zap.String("action", "load metrics common"))
+			time.Sleep(time.Duration(a.pollInterval) * time.Second)
+		}
+	}
+
+}
+
+// Add one more gorutine
+func (a *Agent) RunPollv2(ctx context.Context, wg *sync.WaitGroup) {
+	defer wg.Done()
+	for {
+		select {
+		case <-ctx.Done():
+			logger.Log.Debug("worker is shutdown", zap.String("name", "pollv2"))
+			return
+		default:
+			optData := a.loadPSUtilsMetrics()
+			if a.rateLimit > 0 {
+				logger.Log.Debug("Acquire", zap.String("worker", "pollv2"))
+				a.semaphore.Acquire()
+			}
+			if err := a.sendAllDataOnServer(optData); err != nil {
+				logger.Log.Warn(err.Error())
+			}
+			if a.rateLimit > 0 {
+				logger.Log.Debug("Release", zap.String("worker", "pollv2"))
+				a.semaphore.Release()
+			}
+			logger.Log.Debug("message from worker", zap.String("name", "poll"), zap.String("action", "load metrics optionally"))
 			time.Sleep(time.Duration(a.pollInterval) * time.Second)
 		}
 	}
 
 }
 
+func (a *Agent) loadPSUtilsMetrics() Data {
+	v, err := mem.VirtualMemory()
+	if err != nil {
+		logger.Log.Warn(err.Error())
+		return Data{}
+	}
```

Comment:
Мне кажется об отсутствии значения проще сигнализировать посредством возврата `nil`, а не пустой структуры. `nil` проще интерпретировать. 
```suggestion
func (a *Agent) loadPSUtilsMetrics() *Data {
	v, err := mem.VirtualMemory()
	if err != nil {
		logger.Log.Warn(err.Error())
		return nil
	}
```